src/services/agents/judge_agent.py

```
@@ -138,28 +141,75 @@ def parse_situation(state: JudgeAgentState) -> dict:
+            model="gpt-4o",
+            messages=[
+                {"role": "system", "content": system_prompt},
+                {"role": "user", "content": situation},
+            ],
+        )
+        keywords = (response.choices[0].message.content or "").strip()
+        if not keywords:
+            raise ValueError("Empty keyword response")
+        logger.info(f"Extracted search keywords: {keywords}")
+        return {"search_query": keywords}
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)